routers/web/org/setting.go

```diff
@@ -130,7 +129,9 @@ func SettingsPost(ctx *context.Context) {
 				return
 			}
 		}
-	} else if nameChanged {
+	}
+
+	if nameChanged {
 		if err := repo_model.UpdateRepositoryOwnerNames(org.ID, org.Name); err != nil {
```

Comment:
https://github.com/go-gitea/gitea/blob/491cc06ffe3491242ad9ff6227423d99e673d0c2/services/org/org.go#L72-L74
`RenameOrganization` call `RenameUser`, and  in `RenameUser` we will call `UpdateRepositoryOwnerNames`, so it is no need to call it again here?

Reply:
`RenameOrganization` -> `RenameUser` -> `UpdateRepositoryOwnerNames`, Why call it again?

Reply:
> `RenameOrganization` -> `RenameUser` -> `UpdateRepositoryOwnerNames`, Why call it again?

In L70:
`RenameOrganization` -> `RenameUser` -> `UpdateRepositoryOwnerNames`
`UpdateRepositoryOwnerNames` will be called for the first time.

In L135, `UpdateRepositoryOwnerNames` will be called again.

So `UpdateRepositoryOwnerNames` will be called two times.

Reply:
Repeatedly called it

Reply:
Removed the second call

Reply:
Removed?

Reply:
not removed

Reply:
now file changed:
```diff
diff --git a/routers/web/org/setting.go b/routers/web/org/setting.go
index b5653160a..4e9cba355 100644
--- a/routers/web/org/setting.go
+++ b/routers/web/org/setting.go
@@ -89,7 +89,6 @@ func SettingsPost(ctx *context.Context) {
                // reset ctx.org.OrgLink with new name
                ctx.Org.OrgLink = setting.AppSubURL + "/org/" + url.PathEscape(form.Name)
                log.Trace("Organization name changed: %s -> %s", org.Name, form.Name)
-               nameChanged = false
        }
 
        // In case it's just a case change.
@@ -130,7 +129,9 @@ func SettingsPost(ctx *context.Context) {
                                return
                        }
                }
-       } else if nameChanged {
+       }
+
+       if nameChanged {
                if err := repo_model.UpdateRepositoryOwnerNames(org.ID, org.Name); err != nil {
                        ctx.ServerError("UpdateRepository", err)
                        return
diff --git a/services/user/user.go b/services/user/user.go
index e0815bd86..bb3dd002e 100644
--- a/services/user/user.go
+++ b/services/user/user.go
@@ -58,7 +58,7 @@ func RenameUser(ctx context.Context, u *user_model.User, newUserName string) err
                        u.Name = oldUserName
                        return err
                }
-               return nil
+               return repo_model.UpdateRepositoryOwnerNames(u.ID, newUserName)
        }
 
        ctx, committer, err := db.TxContext(ctx)
```

Reply:
> Removed?

yes